### DISCUSSION TITLE
phpipam + nginx-proxy = connection refused

### DISCUSSION THREAD

**Discussion by Leniwcowaty** [nginx-proxy/nginx-proxy#2647] [Q&A]
Hello there.
I have setup of 4 Docker projects:
- Nginx-proxy
- Heimdall
- Privatebin
- PHPIpam

I decided to use nginx-proxy to provide SSL and reverse-proxy to my services. All of them have in **docker-compose.yml** lines:
`- VIRTUAL_HOST=<service>.<domain>.<tld>`
`- VIRTUAL_PORT=<port>`
`- VIRTUAL_PATH=/`

(of course stuff in <> is properly configured)

As well as (including nginx-proxy):
`network_mode: proxy-network`

With Heimdall and Privatebin it works like a charm - SSL is being applied, reverse proxy works. Only when I get to PHPIpam I get **502** error. When looking at `docker logs nginx-proxy` I see this error:

`nginx.1     | 2023/09/26 13:26:49 [error] 35#35: *1 connect() failed (111: Connection refused) while connecting to upstream, client: <client IP>, server: phpipam.<domain>.<tld>, request: "GET / HTTP/2.0", upstream: "http://<docker network IP>:80/", host: "phpipam.<domain>.<tld>"
nginx.1     | phpipam.<domain>.<tld> <client IP> - - [26/Sep/2023:13:26:49 +0000] "GET / HTTP/2.0" 502 157 "-" "curl/7.81.0" "<docker network IP>:80"`

So it looks like PHPIpam is refusing connection? But is this even possible in Docker? I tried with and without UFW, multiple subdomains. Nothing helps.
Please provide me with some info on how I can resolve this issue...

OS: Ubuntu Server 22.04.3
Docker version: 24.0.6 (from apt)

**Comment by illegalPointer** [comment]
Try the envvar - `IPAM_TRUST_X_FORWARDED=true` on the PHPIPam container.
